processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.Rmd

```diff
@@ -0,0 +1,122 @@
+importable_data_dir <- here("processing-and-analysis", "01-importable-data", "/")
+#add analysis directory string
+analysis_data_dir <- here("processing-and-analysis", "03-analysis-data", "/")
+metadata_dir <- here("original-data", "metadata", "/")
+```
+### load the data set through here directory
+```{r}
+# CFC: Copy rds file from the `original_data_dir` to the `importable_data_dir`.
+copy_from <- paste0(original_data_dir, "master_2s_small_deid_scaled")
+copy_to <- paste0(importable_data_dir, "master_2s_small_deid_scaled")
+# CFC: If the file already exists in the target directory, the file will not be 
+# overwritten and this command will return `FALSE`.
+file.copy(copy_from, copy_to)
+```
+###Import and read the data
+```{r}
+#data_2s <- readRDS(paste0(metadata_dir, "master_2s_small_deid_scaled.rds"))
+data_2s <- readRDS("G:/Shared drives/CEG Two-Stage Exams Analysis/Ganling/two-stage-2/original-data/master_2s_small_deid_scaled.rds")
+```
+##Data wrangling
+####Creating unique rows and remove the ones that are not necessary for this data model
+```{r}
+data_2s_extras_removed = subset(data_2s, select = -c(item_num, stud_ans, exam_key, corr, bloom_rating, complexity_rating_mean, complexity_rating_median, item_code, stem))
+data_2s_unique = unique(data_2s_extras_removed, incomparables = FALSE)
+View(data_2s_unique)
+data_2s_exam_score <- data_2s_unique %>% 
+  select(exam1, exam2, finalexam, course_fullid, ver, sex_id, urm_id, eop_id, fgn_id, experiment1, ta_sect, satverbal, satmath, high_sch_gpa, mastered_topics_initial_kc)
+  
+data_2s_exam_score_true <- na.omit(data_2s_exam_score)
+```
+##Make 'ta_sect'unique across different data values
+####First, we need to split the data from into year 16 and year 17 
+```{r}
+data_2s_exam_score_true$ta_sect <- as.character(data_2s_exam_score_true$ta_sect)
+
+#next, split the dataframe in half
+master_unique_2016 <- subset(data_2s_exam_score_true, course_fullid != "CHEM_142_A_2017_4" & course_fullid != "CHEM_142_B_2017_4")
+master_unique_2017 <- subset(data_2s_exam_score_true, course_fullid != "CHEM_142_A_2016_4" & course_fullid != "CHEM_142_B_2016_4")
+```
+####Use the vectorization function in R, we can add '_16' and '_17'at the end of each TA section 
+```{r}
+master_unique_2016$ta_sect <- paste0(master_unique_2016$ta_sect, "_16")
+master_unique_2017$ta_sect <- paste0(master_unique_2017$ta_sect, "_17")
+```
+##recombine the dataframes together and make the 'ta_sect' factors again
+```{r}
+master_true <- rbind(master_unique_2016, master_unique_2017)
+master_true$ta_sect <- as.factor(master_true$ta_sect)
+```
+####I would like to create a model that could implement the random effects based on the TA sections
+```{r}
+#First, include sex_id as a control variable
+mod1 <- lmer(finalexam ~ experiment1 + satmath + satverbal + mastered_topics_initial_kc + high_sch_gpa + sex_id + (1|ta_sect), 
+           data=master_true)
+summary(mod1)
+AIC(mod1)
+
+#Include eop_id as a control variable
```

Comment:
`eop_id` and other demographics variables aren't really controls in our work... they are groups designated by categorical variables that we care about in our models. We are trying to compare groups of interest (women, URMs, FGNs, EOPs) to their traditionally overserved counterparts. This is in contrast to the control variables which we want to include to make sure we compare students at the same "elevation"--- across all demo groups--- in terms of academic backgrounds, but we don't really care about the estimates that come out of the regression for the control variables. We do care about the outcomes for the groups of interest. We can discuss this more tomorrow.